core/src/test/java/org/infinispan/persistence/file/SingleFileStoreCompatibilityTest.java

```diff
@@ -0,0 +1,89 @@
+package org.infinispan.persistence.file;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import org.infinispan.commons.test.CommonsTestingUtil;
+import org.infinispan.commons.util.FileLookupFactory;
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.persistence.PersistenceCompatibilityTest;
+import org.testng.annotations.Test;
+
+/**
+ * Tests if {@link SingleFileStore} can migrate data from Infinispan 10.1.x.
+ *
+ * @author Pedro Ruivo
+ * @since 11.0
+ */
+@Test(groups = "functional", testName = "persistence.file.SingleFileStoreCompatibilityTest")
+public class SingleFileStoreCompatibilityTest extends PersistenceCompatibilityTest<String> {
+
+   private static final String DATA_10_1 = "10_1_x_sfs_data/sfs-store-cache.dat";
+   private String tmpDirectory;
```

Comment:
We can pass this to the constructor of `PersistenceCompatibilityTest` so that we don't have to override `setup` and `teardown` in all of the tests.

Reply:
:+1: